dashboard/client/src/components/PlacementGroupTable.tsx

```diff
@@ -0,0 +1,155 @@
+import Pagination from "@material-ui/lab/Pagination";
+import React, { useState } from "react";
+import rowStyles from "../common/RowStyles";
+import { PlacementGroup } from "../type/placementGroup";
+import { useFilter } from "../util/hook";
+import StateCounter from "./StatesCounter";
+import { StatusChip } from "./StatusChip";
+
+const PlacementGroupTable = ({
+  placementGroups = [],
+  jobId = null,
+}: {
+  placementGroups: Array<PlacementGroup>;
+  jobId?: string | null;
+}) => {
+  const [pageNo, setPageNo] = useState(1);
+  const { changeFilter, filterFunc } = useFilter();
+  const [pageSize, setPageSize] = useState(10);
+  const placementGroupList = placementGroups.filter(filterFunc);
+  const list = placementGroupList.slice(
+    (pageNo - 1) * pageSize,
+    pageNo * pageSize,
+  );
+  const classes = rowStyles();
+
+  const columns = [
+    { label: "ID" },
+    { label: "Name" },
+    { label: "Job Id" },
+    { label: "State" },
+    { label: "Scheduling Detail" },
+  ];
+
+  return (
+    <React.Fragment>
+      <div style={{ flex: 1, display: "flex", alignItems: "center" }}>
+        <Autocomplete
+          style={{ margin: 8, width: 120 }}
+          options={Array.from(
+            new Set(placementGroups.map((e) => e.placement_group_id)),
+          )}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("state", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Placement group ID" />
+          )}
+        />
+        <Autocomplete
+          style={{ margin: 8, width: 150 }}
+          defaultValue={jobId}
+          options={Array.from(
+            new Set(placementGroups.map((e) => e.creator_job_id)),
+          )}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("creator_job_id", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Job Id" />
+          )}
+        />
+        <Autocomplete
+          style={{ margin: 8, width: 150 }}
+          options={Array.from(new Set(placementGroups.map((e) => e.name)))}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("name", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Name" />
+          )}
+        />
+        <TextField
+          style={{ margin: 8, width: 120 }}
+          label="Page Size"
+          size="small"
+          defaultValue={10}
+          InputProps={{
+            onChange: ({ target: { value } }) => {
+              setPageSize(Math.min(Number(value), 500) || 10);
+            },
+            endAdornment: (
+              <InputAdornment position="end">Per Page</InputAdornment>
+            ),
+          }}
+        />
+      </div>
+      <div style={{ display: "flex", alignItems: "center" }}>
```

Comment:
I know this is existing pattern but direct inlining styles is frowned upon because of low composibility.

I hoped we could go in polish each page and rewrite with better code style before adding too much new content.

Reply:
What does it take to do that? I mostly copied & pasted existing actor table (and I feel bad about it). Alternative idea is to do clean up as a part of "cleaning tech debt" after 2.3. I feel like the amount of code we have is small enough that we can proceed this way.

Reply:
for this part, I will avoid inlined style... is the preferred style using classes? Or do we have a doc for the frontend code guideline? (if so can you share it? )

Reply:
hmm I found refactoring this is actually much bigger efforts that I thought. Let's refactor all at once?